AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -165,6 +167,7 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
             let metadata = MutationSyncMetadata.keys
             storageEngine.query(MutationSyncMetadata.self,
                                 predicate: metadata.id == model.id,
+                                sort: nil,
```

Comment:
sorting doesn't make sense in a method that just queries for 1 model by its id.

Reply:
Oh, here just need to make compiler happy, like this one: https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/AWSMutationDatabaseAdapter/AWSMutationDatabaseAdapter%2BMutationEventSource.swift#L32
Or maybe I don't get your meaning here

Reply:
You should update the storage engine method signature to make `sort` default to `nil`, rather than requiring it to be specified as `nil` at the call site